tests/ci/ci.py

```diff
@@ -0,0 +1,665 @@
+    print(f"CI docs metadata files [{files_docs}]")
+    files += files_docs
+
+    # parse CI metadata
+    job_digests = indata["jobs_data"]["digests"]
+    # create GH status
+    pr_info = PRInfo()
+    commit = get_commit(Github(get_best_robot_token(), per_page=100), pr_info.sha)
+
+    def run_create_status(job, digest, batch, num_batches):
+        success_flag_name = get_file_flag_name(job, digest, batch, num_batches)
+        if success_flag_name in files:
+            print(f"Going to re-create GH status for job [{job}] sha [{pr_info.sha}]")
+            job_status = CommitStatusData.load_from_file(
+                f"{TEMP_PATH}/{success_flag_name}"
+            )  # type: CommitStatusData
+            assert job_status.status == "success", "BUG!"
+            commit.create_status(
+                state=job_status.status,
+                target_url=job_status.report_url,
+                description=f"Reused from [{job_status.pr_num}-{job_status.sha[0:8]}]: {job_status.description}",
+                context=get_check_name(job, batch=batch, num_batches=num_batches),
+            )
+            print(f"GH status re-created from file [{success_flag_name}]")
+
+    with concurrent.futures.ThreadPoolExecutor() as executor:
+        futures = []
+        for job in job_digests:
+            if is_build_job(job):
+                # no GH status for build jobs
+                continue
+            digest = job_digests[job]
+            num_batches = CI_CONFIG.get_job_config(job).num_batches
+            for batch in range(num_batches):
+                future = executor.submit(
+                    run_create_status, job, digest, batch, num_batches
+                )
+                futures.append(future)
+        done, _ = concurrent.futures.wait(futures)
+        for future in done:
+            try:
+                _ = future.result()
+            except Exception as e:
+                raise e
+    print("Going to update overall CI report")
+    set_status_comment(commit, pr_info)
+    print("... CI report update - done")
+
+    # clean up
+    ci_files = list(temp_path.glob("*.ci"))
+    for file in ci_files:
+        file.unlink()
+
+
+def main() -> int:
+    exit_code = 0
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    args = parse_args(parser)
+
+    if args.mark_success or args.pre or args.post or args.run:
+        assert args.infile, "Run config must be provided via --infile"
+        assert args.job_name, "Job name must be provided via --job-name"
+
+    indata: Optional[Dict[str, Any]] = None
+    if args.infile:
+        indata = (
+            json.loads(args.infile)
+            if not os.path.isfile(args.infile)
+            else json.load(open(args.infile))
+        )
+        assert indata and isinstance(indata, dict), "Invalid --infile json"
+
+    result: Dict[str, Any] = {}
+    s3 = S3Helper()
+
+    if args.configure:
+        GR = GitRunner()
```

Comment:
Why do you create another instance instead of using `git_helper.git_runner`?

btw, `GR()` and `GR.run()` are the same